============================= test session starts ==============================
platform linux -- Python 3.8.10, pytest-6.2.5, py-1.11.0, pluggy-1.0.0 -- /usr/bin/python3
cachedir: .pytest_cache
rootdir: /home/yanghong/Projects/SQLProject
plugins: parallel-0.1.1
collecting ... collected 0 items / 1 error

==================================== ERRORS ====================================
________________________ ERROR collecting test session _________________________
/usr/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1014: in _gcd_import
    ???
<frozen importlib._bootstrap>:991: in _find_and_load
    ???
<frozen importlib._bootstrap>:975: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:671: in _load_unlocked
    ???
../../.local/lib/python3.8/site-packages/_pytest/assertion/rewrite.py:170: in exec_module
    exec(co, module.__dict__)
fe/conftest.py:4: in <module>
    from bookstore import serve
bookstore/serve.py:1: in <module>
    from flask import Flask
../../.local/lib/python3.8/site-packages/flask/__init__.py:3: in <module>
    from werkzeug.exceptions import abort as abort
../../.local/lib/python3.8/site-packages/werkzeug/__init__.py:1: in <module>
    from .serving import run_simple as run_simple
../../.local/lib/python3.8/site-packages/werkzeug/serving.py:31: in <module>
    from .exceptions import InternalServerError
../../.local/lib/python3.8/site-packages/werkzeug/exceptions.py:692: in <module>
    class _RetryAfter(HTTPException):
../../.local/lib/python3.8/site-packages/werkzeug/exceptions.py:702: in _RetryAfter
    retry_after: t.Optional[t.Union[datetime, int]] = None,
/usr/lib/python3.8/typing.py:258: in inner
    return cached(*args, **kwds)
/usr/lib/python3.8/typing.py:365: in __getitem__
    return Union[arg, type(None)]
/usr/lib/python3.8/typing.py:258: in inner
    return cached(*args, **kwds)
/usr/lib/python3.8/typing.py:358: in __getitem__
    parameters = tuple(_type_check(p, msg) for p in parameters)
/usr/lib/python3.8/typing.py:358: in <genexpr>
    parameters = tuple(_type_check(p, msg) for p in parameters)
/usr/lib/python3.8/typing.py:141: in _type_check
    arg.__origin__ in invalid_generic_forms):
/usr/lib/python3.8/typing.py:327: in __eq__
    return self._name == other._name
../../.local/lib/python3.8/site-packages/coverage/pytracer.py:178: in _trace
    if self.trace_arcs and self.cur_file_data:
E   KeyboardInterrupt
=========================== short test summary info ============================
ERROR  - KeyboardInterrupt
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
=============================== 1 error in 1.06s ===============================
No data to combine
Name                            Stmts   Miss Branch BrPart  Cover
-----------------------------------------------------------------
bookstore/Token.py                 33     33     12      0     0%
bookstore/__init__.py               0      0      0      0   100%
bookstore/bp/__init__.py            0      0      0      0   100%
bookstore/bp/auth.py               74     74      8      0     0%
bookstore/bp/buyer.py              73     73      6      0     0%
bookstore/bp/extra.py              62     62      4      0     0%
bookstore/bp/seller.py             77     77     12      0     0%
bookstore/classes/__init__.py       0      0      0      0   100%
bookstore/classes/model.py          4      4      0      0     0%
bookstore/classes/order.py         39     39     10      0     0%
bookstore/classes/searcher.py      29     29     16      0     0%
bookstore/classes/shop.py          42     42     16      0     0%
bookstore/classes/sql.py           63     63     20      0     0%
bookstore/classes/thread.py        38     38     19      0     0%
bookstore/classes/user.py         114    114     52      0     0%
bookstore/db_handler.py            41     41      8      0     0%
bookstore/error.py                 64     64     34      0     0%
bookstore/serve.py                 31     30      4      0     3%
fe/__init__.py                      0      0      0      0   100%
fe/access/__init__.py               0      0      0      0   100%
fe/access/auth.py                  31     31      2      0     0%
fe/access/book.py                  70     70     16      0     0%
fe/access/buyer.py                 50     50      4      0     0%
fe/access/new_buyer.py              8      8      0      0     0%
fe/access/new_seller.py             8      8      0      0     0%
fe/access/seller.py                37     37      2      0     0%
fe/bench/__init__.py                0      0      0      0   100%
fe/bench/run.py                    13     13      6      0     0%
fe/bench/session.py                47     47     14      0     0%
fe/bench/workload.py              125    125     28      0     0%
fe/conf.py                         11     11      0      0     0%
fe/conftest.py                     17     13      0      0    24%
-----------------------------------------------------------------
TOTAL                            1201   1196    293      0     1%
Wrote HTML report to htmlcov/index.html
